### PR TITLE
Feature/member cache

### LINKED ISSUE #207: cache members for cut validation
To prevent sql injection, validate cut members against cached member list. Can use current logic layer cache.

### PR DESCRIPTION
**Pull request by hwchen**
This is a basic pass at using a members cache to be able to validate cut member IDs before sending it on to sql.

This needs some work before being robust enough to work
- without any logic layer involvement
- for a /members endpoint

However, in its current form, it may be usable enough for our requirements.

```
TODO note that this is being used in core tesseract, but that
the cache is created using logic layer rules. This means that
at the moment of this implementation, this will work in core
tesseract but only if the core tesseract schema can also be a
logic layer schema (and satisfying all the rules about level,
property uniqueness, etc.) The next step is to figure out how
to architecture this cache so that it can be used even in
non-logic layer setups, but also be used for members endpoint
(which requires ID)
```

closes #207

### CODE REVIEW

File: tesseract-server/src/logic_layer/cache.rs [hunk 1]
```diff
@@ -225,6 +225,19 @@ impl CubeCache {
             None => None
         }
     }
+
+    // TODO note that this is being used in core tesseract, but that the cache is created using
+    // logic layer rules. This means that at the moment of this implementation, this will work in
+    // core tesseract but only if the core tesseract schema can also be a logic layer schema (and
+    // satisfying all the rules about level, property uniqueness, etc.)
```
Comment (jspeis) [inline tesseract-server/src/logic_layer/cache.rs]: overall this looks great! what will happen if a user doesn't satisfy the rules about level/property uniqueness at the moment, what kind of error would they get?

**Comment by jspeis** [comment]
@MarcioPorto I've pushed some minor tweaks, when you have a moment, can you take a look to see if this looks good to go?

File: tesseract-server/src/logic_layer/cache.rs [hunk 2]
```diff
@@ -425,8 +447,8 @@ pub fn populate_cache(
         })
     }
 
-    info!("Cache ready!");
-
+    let timing = time_start.elapsed();
```
Comment (MarcioPorto) [inline tesseract-server/src/logic_layer/cache.rs]: This is great. Thanks for adding it.

File: tesseract-server/src/handlers/aggregate.rs [hunk 3]
```diff
@@ -240,3 +265,21 @@ impl TryFrom<AggregateQueryOpt> for TsQuery {
         })
     }
 }
+
+use failure::{bail, format_err};
+use tesseract_core::names::Cut;
+use crate::logic_layer::CubeCache;
+
+fn validate_members(cuts: &[Cut], cube_cache: &CubeCache) -> Result<(), Error> {
```
Comment (MarcioPorto) [inline tesseract-server/src/handlers/aggregate.rs]: Should we move this somewhere where to logic layer could use it as well?

Comment (jspeis) [reply tesseract-server/src/handlers/aggregate.rs]: Agreed! What do you think about having it in `handlers/util.rs`? (see latest push)

Comment (MarcioPorto) [reply tesseract-server/src/handlers/aggregate.rs]: That looks good to me!

**Comment by MarcioPorto** [comment]
@jspeis Made a few changes to also include this check to the logic layer aggregate method.

File: tesseract-server/src/logic_layer/cache.rs (hunk 1 above)
Comment (jspeis) [reply tesseract-server/src/logic_layer/cache.rs]: @MarcioPorto we should just review this together

**Pull request merged**